src/Form/MessageTypeForm.php

```diff
@@ -130,6 +130,13 @@ public function form(array $form, FormStateInterface $form_state) {
       '#states' => $states,
     );
 
+    $form['settings']['token options']['token replace'] = array(
+      '#title' => t('Token replace'),
+      '#type' => 'checkbox',
+      '#description' => t('When this option is selected, token processing will happen.'),
+      '#default_value' => isset($settings['token options']['token replace']) ? $settings['token options']['token replace'] : TRUE,
```

Comment:
enough just to do `'#default_value' =>  !empty($settings['token options']['token replace'])`